mlflow/models/docker_utils.py

```diff
@@ -96,6 +96,9 @@ def _get_maven_proxy():
          maven \
     && rm -rf /var/lib/apt/lists/*
 
+# Increase default timeout for API throttling
+ENV PIP_DEFAULT_TIMEOUT=100
```

Comment:
Why do we need this change?

Reply:
While I was debugging failures in that draft branch (and some PR test failures over the last few days, I noticed a few pip install failures with timeout exceptions from the package repository. These intermittent failures generally resolve themselves but the default timeout doesn't allow for enough total time with the 5x retry policy within `pip install`. I checked a few discussion boards where people were seeing these intermittent failures in VMs that are building multiple Docker containers for CI and the general consensus was to use a container-level global config for pip (not in the main CI) of overriding https://pip.pypa.io/en/stable/topics/configuration/?highlight=timeout#environment-variables . If we don't want to do this, I'll remove it :)